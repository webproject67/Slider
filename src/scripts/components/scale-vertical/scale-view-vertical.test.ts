import SliderModel from '../model/slider-model';
import ScaleViewVertical from './scale-view-vertical';

const main = document.createElement('div');
const state = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 0,
  range: 'one',
  scale: true,
  step: 1,
  to: -10000,
  toPercent: 100,
  view: 'vertical'
};

const model = new SliderModel(main, state);
const scale = new ScaleViewVertical(model);

test('spyOn click scale vertical', () => {
  const somethingSpy = jest.spyOn(scale, 'handleItemClick');
  let evt: any;
  scale.handleItemClick(evt);
  expect(somethingSpy).toHaveBeenCalledTimes(1)
});

it('jest snapshots from element', () => {
  expect(scale.element).toMatchSnapshot()
})

it('jest snapshots from new element', () => {
  expect(scale.newElement).toMatchSnapshot()
})

const otherMain = document.createElement('div');
const otherState = {
  flag: true,
  from: -10000,
  fromPercent: 0,
  max: 100,
  min: 1,
  range: 'one',
  scale: true,
  step: 2,
  to: -10000,
  toPercent: 100,
  view: 'vertical'
};

const otherModel = new SliderModel(otherMain, otherState);
const otherScale = new ScaleViewVertical(otherModel);

it('jest snapshots from other element', () => {
  expect(otherScale.element).toMatchSnapshot()
})