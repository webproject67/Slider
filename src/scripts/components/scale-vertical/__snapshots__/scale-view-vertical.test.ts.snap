// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest snapshots 1`] = `
<div
  class="slider__list slider__list--transform"
>
  
          
  <div
    class="slider__item"
  >
    |
            
    <span
      class="slider__item--number slider__item--max"
    >
      100
    </span>
    
          
  </div>
  
        
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  <div
    class="slider__item"
  >
    |
  </div>
  
          
  <div
    class="slider__item"
    data-testid="scaleVertical"
  >
    |
            
    <span
      class="slider__item--number slider__item--min"
    >
      0
    </span>
    
          
  </div>
  
        
</div>
`;
